repos/spack_repo/builtin/packages/py_dolfinx_mpc/package.py

```diff
@@ -0,0 +1,47 @@
+from spack_repo.builtin.build_systems.python import PythonPackage
+
+from spack.package import *
+
+
+class PyDolfinxMpc(PythonPackage):
+    """Python interface of DOLFINx-MPC, an extension of DOLFINx for multipoint constraints."""
+
+    homepage = "https://www.jsdokken.com/dolfinx_mpc"
+    url = "https://github.com/jorgensd/dolfinx_mpc/archive/v0.9.3.tar.gz"
+    git = "https://github.com/jorgensd/dolfinx_mpc.git"
+
+    maintainers("jorgensd")
+
+    license("MIT", checked_by="jorgensd")
+
+    version("main", branch="main")
+    version("0.9.3", sha256="efa312cc498e428aab44acccc9bb0c74c200eda005742de7778c8e68fa84e8df")
+    version("0.8.1", sha256="e0254b4a1c9c1456583c1415821946b11b0b2e48dbfee6558da2bbedfe78b461")
+
+    variant("numba", default=False, description="numba support")
+
+    depends_on("cxx", type="build")
+
+    depends_on("cmake@3.21:", when="@0.9:", type="build")
+    depends_on("cmake@3.19:", when="@:0.8", type="build")
+    depends_on("py-packaging")
+
+    depends_on("python@3.9:", when="@0.8:", type=("build", "run"))
```

Comment:
Versions `0.8.1` and `0.9.3` require `python@3.8:` per https://github.com/jorgensd/dolfinx_mpc/blob/v0.8.1/python/pyproject.toml and https://github.com/jorgensd/dolfinx_mpc/blob/v0.9.3/python/pyproject.toml.